test/unit/mc/test_triangular_solve_local.cpp

```diff
@@ -0,0 +1,337 @@
+
+    double i = index.row();
+    double j = index.col();
+
+    T gamma = TypeUtilities<T>::polar((i + 1) / (j + 2), 2 * i + j);
+
+    if (diag == blas::Diag::Unit)
+      return ((kk - 1) * gamma + res_b(index)) / alpha;
+    else
+      return kk * gamma / alpha;
+  };
+
+  return std::make_tuple(el_op_a, el_b, res_b);
+}
+
+/// @brief Returns el_op_a, el_b, res_b for side = Right. Same implementation as in test_trsm.h (there
+/// referred to tiles).
+template <class T>
+auto testTriangularSolveElementFunctionsRight(blas::Uplo uplo, blas::Op op, blas::Diag diag, T alpha,
+                                              SizeType n) {
+  // Note: The tile elements are chosen such that:
+  // - res_ik = (k+.5) / (i+2) * exp(I*(i+k)),
+  // - op(a)_kj = (j+1) / (k+.5) * exp(I*(2*j-k)) for the referenced elements
+  //   op(a)_kj = -9.9 otherwise,
+  // where I = 0 for real types or I is the complex unit for complex types.
+  // Therefore the elements of b should be:
+  // b_ij = (Sum_k res_ik * op(a)_kj) / alpha
+  //      = (res_ij * op(a)_jj + (kk-1) * gamma) / alpha,
+  // where gamma = (j+1) / (i+2) * exp(I*(i+2*j)),
+  //       kk = j+1 if op(a) is an upper triangular matrix, or
+  //       kk = n-j if op(a) is an lower triangular matrix.
+  // Therefore
+  // b_ij = (res_ij + (kk-1) * gamma) / alpha, if diag == Unit
+  // b_ij = kk * gamma / alpha, otherwise.
+
+  bool op_a_lower = false;
+  if ((uplo == blas::Uplo::Lower && op == blas::Op::NoTrans) ||
+      (uplo == blas::Uplo::Upper && op != blas::Op::NoTrans))
+    op_a_lower = true;
+
+  auto res_b = [](const GlobalElementIndex& index) {
+    double i = index.row();
+    double k = index.col();
+
+    return TypeUtilities<T>::polar((k + .5) / (i + 2), i + k);
+  };
+
+  auto el_op_a = [op_a_lower, diag](const GlobalElementIndex& index) {
+    if ((op_a_lower && index.row() < index.col()) || (!op_a_lower && index.row() > index.col()) ||
+        (diag == blas::Diag::Unit && index.row() == index.col()))
+      return TypeUtilities<T>::element(-9.9, 0);
+
+    double k = index.row();
+    double j = index.col();
+
+    return TypeUtilities<T>::polar((j + 1) / (k + .5), 2 * j - k);
+  };
+
+  auto el_b = [n, alpha, diag, op_a_lower, res_b](const GlobalElementIndex& index) {
+    BaseType<T> kk = op_a_lower ? n - index.col() : index.col() + 1;
+
+    double i = index.row();
+    double j = index.col();
+    T gamma = TypeUtilities<T>::polar((j + 1) / (i + 2), i + 2 * j);
+
+    if (diag == blas::Diag::Unit)
+      return ((kk - 1) * gamma + res_b(index)) / alpha;
+    else
+      return kk * gamma / alpha;
+  };
+
+  return std::make_tuple(el_op_a, el_b, res_b);
+}
+
+template <class T>
+void testTriangularSolve(blas::Side side, blas::Uplo uplo, blas::Op op, blas::Diag diag, T alpha,
+                         Matrix<T, Device::CPU>& matA, Matrix<T, Device::CPU>& matB,
+                         Matrix<T, Device::CPU>& matX) {
```

Comment:
I would leave to this function the creation of the matrix.
This function arguments would be `side, uplo, op, diag, m, n, mb, nb`

Reply:
The function now has arguments: `side, uplo, op, diag, size, colB, block_size`. What do you think? Can I improve it?

Reply:
See new comment.